### PR TITLE
Note the default GC latency mode

### LINKED ISSUE #9739: Not clear what the default mode for Server GC is
.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 1fcebd46-8a64-9531-f6e1-3a330b61bb0f
* Version Independent ID: 7538a252-8fa8-4615-92df-f1c6d7d6a3e4
* Content: [Latency Modes](https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/latency)
* Content Source: [docs/standard/garbage-collection/latency.md](https://github.com/dotnet/docs/blob/master/docs/standard/garbage-collection/latency.md)
* Product: **dotnet**
* GitHub Login: @rpetrusha
* Microsoft Alias: **ronpet**

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #9739 .

### CODE REVIEW

**Comment by gewarren** [comment]
@tdykstra Could you take a look since Bill is out?

File: docs/standard/garbage-collection/latency.md [hunk 1]
```diff
@@ -7,42 +7,50 @@ helpviewer_keywords:
...
+# Latency modes
 
- Latency refers to the time that the garbage collector intrudes in your application. During low latency periods, the garbage collector is more conservative and less intrusive in reclaiming objects. The <xref:System.Runtime.GCLatencyMode?displayProperty=nameWithType> enumeration provides two low latency settings:
+To reclaim objects, the garbage collector (GC) must stop all the executing threads in an application. *Latency* refers to the time that the garbage collector intrudes in your application. In some situations, such as when an application retrieves data or displays content, a full garbage collection can occur at a critical time and impede performance. You can adjust the intrusiveness of the garbage collector by setting the <xref:System.Runtime.GCSettings.LatencyMode%2A?displayProperty=nameWithType> property to one of the <xref:System.Runtime.GCLatencyMode?displayProperty=nameWithType> values.
```
Comment (tdykstra) [inline docs/standard/garbage-collection/latency.md]: The "Latency refers to ..." sentence seems vague to me. I think your definition of "Low latency" under the following H2 is clearer and makes this sentence unnecessary.

File: docs/standard/garbage-collection/latency.md [hunk 2]
```diff
@@ -7,42 +7,50 @@ helpviewer_keywords:
...
+The <xref:System.Runtime.GCLatencyMode?displayProperty=nameWithType> enumeration provides two low latency settings:
+
+- [GCLatencyMode.LowLatency](xref:System.Runtime.GCLatencyMode.LowLatency) suppresses generation 2 collections and performs only generation 0 and 1 collections. It can be used only for short periods of time. Over longer periods, if the system is under memory pressure, the garbage collector will trigger a collection, which can briefly pause the application and disrupt a time-critical operation. This setting is available only for workstation garbage collection.
+
+- [GCLatencyMode.SustainedLowLatency](xref:System.Runtime.GCLatencyMode.SustainedLowLatency) suppresses foreground generation 2 collections and performs only generation 0, 1, and background generation 2 collections. It can be used for longer periods of time, and is available for both workstation and server garbage collection. This setting cannot be used if background garbage collection is disabled by the .NET Framework [\<gcConcurrent> configuration element](../../../docs/framework/configure-apps/file-schema/runtime/gcconcurrent-element.md).
```
Comment (tdykstra) [inline docs/standard/garbage-collection/latency.md]: Given that this is a .NET standard article, is the reference to .NET Framework appropriate here, i.e., are we saying that LowLatency works cross-platform but SustainedLowLatency only works on Framework?

File: docs/standard/garbage-collection/latency.md [hunk 3]
```diff
@@ -7,42 +7,50 @@ helpviewer_keywords:
...
-|<xref:System.Runtime.GCLatencyMode.SustainedLowLatency>|For applications that have time-sensitive operations for a contained but potentially longer duration of time during which interruptions from the garbage collector could be disruptive. For example, applications that need quick response times as market data changes during trading hours.<br /><br /> This mode results in a larger managed heap size than other modes. Because it does not compact the managed heap, higher fragmentation is possible. Ensure that sufficient memory is available.|
+|<xref:System.Runtime.GCLatencyMode.Batch>|For applications that have no user interface (UI) or server-side operations.<br /><br />This is the default mode for both server and workstation garbage collection when background garbage collection is disabled.|
+|<xref:System.Runtime.GCLatencyMode.Interactive>|For most applications that have a UI.<br /><br />This is the default mode for workstation and server garbage collection for a standalone executable that's not hosted.<br /><br />This is the default mode for both server and workstation garbage collection when background garbage collection is enabled.|
```
Comment (tdykstra) [inline docs/standard/garbage-collection/latency.md]: Having the two "This is the default" lines is confusing, especially as they're out of sync in listing server and workstation in different orders. Does the first line now override the second line, i.e., even if background collection isn't enabled it's the default in the line 1 scenario? 

Also, I presume if an app is "hosted," the GC setting of the hosting process takes precedence; if so should we explicitly state that?

**Review by tdykstra — COMMENTED** [review]
A couple of questions.

File: docs/standard/garbage-collection/latency.md [hunk 4]
```diff
@@ -7,42 +7,52 @@ helpviewer_keywords:
...
-|<xref:System.Runtime.GCLatencyMode.SustainedLowLatency>|For applications that have time-sensitive operations for a contained but potentially longer duration of time during which interruptions from the garbage collector could be disruptive. For example, applications that need quick response times as market data changes during trading hours.<br /><br /> This mode results in a larger managed heap size than other modes. Because it does not compact the managed heap, higher fragmentation is possible. Ensure that sufficient memory is available.|
+|<xref:System.Runtime.GCLatencyMode.Batch>|For applications that have no user interface (UI) or server-side operations.<br /><br />When background garbage collection is disabled, this is the default mode for workstation and server garbage collection.|
+|<xref:System.Runtime.GCLatencyMode.Interactive>|For most applications that have a UI.<br /><br />This is the default mode for workstation and server garbage collection. However, if an app is hosted, the garbage collector settings of the hosting process take precedence.|
```
Comment (gewarren) [inline docs/standard/garbage-collection/latency.md]: @Maoni0 Is this sentence accurate? "However, if an app is hosted, the garbage collector settings of the hosting process take precedence."

Comment (Maoni0) [reply docs/standard/garbage-collection/latency.md]: yes that's accurate for the default.

**Review by tdykstra — APPROVED** [review]

**Comment by gewarren** [comment]
@tdykstra Thanks for the review.

**Pull request merged**